src/FileBackedTaskManager.java

```diff
@@ -0,0 +1,216 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.io.IOException;
+import java.io.FileWriter;
+import java.io.Writer;
+import java.nio.file.Files;
+import java.io.FileReader;
+import java.io.Reader;
+import java.io.File;
+
+public class FileBackedTaskManager extends InMemoryTaskManager  {
+
+    private static final String HOME = System.getProperty("user.home");
+    File dir = new File("C:\\Users\\Igor\\java-kanban\\dbTaskManager");
+
+    File dbTaskManager = new File(dir, "dbTaskManager.csv");
+    private int id = 0;
+    private int lastEpicId = 0;
+    private HashMap<Integer, Task> singleTaskDesc = new HashMap<>();
+    private HashMap<Integer, Epic> epicTaskDesc = new HashMap<>();
+    private HashMap<Integer, SubTask> subTaskDesc = new HashMap<>();
+    InMemoryHistoryManager inMemoryHistoryManager = new InMemoryHistoryManager();
+
+    public FileBackedTaskManager(File dbTaskManager, int id, int lastEpicId, HashMap<Integer, Task> singleTaskDesc, HashMap<Integer, Epic> epicTaskDesc, HashMap<Integer, SubTask> subTaskDesc, InMemoryHistoryManager inMemoryHistoryManager) {
+        this.dbTaskManager = dbTaskManager;
+        this.id = id;
+        this.lastEpicId = lastEpicId;
+        this.singleTaskDesc = singleTaskDesc;
+        this.epicTaskDesc = epicTaskDesc;
+        this.subTaskDesc = subTaskDesc;
+        this.inMemoryHistoryManager = inMemoryHistoryManager;
+    }
+
+
+    @Override
+    public Integer getLastEpicId() {
+        return super.getLastEpicId();
+    }
+
+    @Override
+    public Integer genId() {
+        return super.genId();
+    }
+
+
+    @Override
+    public ArrayList<Task> getTasks() {
+        return super.getTasks();
+    }
+
+    @Override
+    public HashMap<Integer, Task> deleteTasks() {
+        return super.deleteTasks();
+    }
+
+    @Override
+    public HashMap<Integer, Epic> deleteEpics() {
+        return super.deleteEpics();
+    }
+
+    @Override
+    public HashMap<Integer, SubTask> deleteSubTasks() {
+        return super.deleteSubTasks();
+    }
+
+    @Override
+    public Task addTask(Task newSingleTask) {
+        super.addTask(newSingleTask);
+        save();
+        return super.addTask(newSingleTask);
+    }
+
+    @Override
+    public Epic addEpic(Epic newEpic) {
+        super.addEpic(newEpic);
+        save();
+        return super.addEpic(newEpic);
+    }
+
+    @Override
+    public SubTask addSubTask(SubTask newSubTask) {
+        super.addSubTask(newSubTask);
+        save();
+        return super.addSubTask(newSubTask);
```

Comment:
Здесь и в других методах нужно избавиться от дублирования
```suggestion
        SubTask subtask = super.addSubTask(newSubTask);
        save();
        return subtask;
```

Reply:
странно - вроде вчера все методы поправил как было предложено. Исправлю